### PR TITLE
Fix #4146 - Airflow REST delete DAG

### LINKED ISSUE #4146: Airflow REST - delete APIs

### PR DESCRIPTION
**Pull request by pmbrull**
### Describe your changes :

Fix #4146

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [x] New feature
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: openmetadata-airflow-apis/src/openmetadata/operations/delete.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    with settings.Session() as session:
+
+        deleted_dags = (
+            session.query(DagModel).filter(DagModel.dag_id == dag_id).delete()
+        )
+        session.commit()
+
+    if deleted_dags > 0 and deleted_file:
+
```
Comment (harshach) [inline openmetadata-airflow-apis/src/openmetadata/operations/delete.py:46]: @pmbrull should this new line needs to be removed?

File: openmetadata-airflow-apis/src/openmetadata/api/apis_metadata.py [hunk 2]
```diff
@@ -72,6 +72,18 @@
             },
         ],
     },
+    {
+        "name": "delete_dag",
+        "description": "Delete a DAG in the Web Server from Airflow database and filesystem",
+        "http_method": "POST",
```
Comment (harshach) [inline openmetadata-airflow-apis/src/openmetadata/api/apis_metadata.py:78]: should we make this DELETE method instead?

Comment (pmbrull) [reply openmetadata-airflow-apis/src/openmetadata/api/apis_metadata.py:78]: that's a good point, let me check and change it

File: openmetadata-airflow-apis/src/openmetadata/operations/delete.py (hunk 1 above)
Comment (pmbrull) [reply openmetadata-airflow-apis/src/openmetadata/operations/delete.py:46]: do you mean the validation or the blank space? Here I'm checking that the delete query returns something and that we could properly delete the file

Comment (harshach) [reply openmetadata-airflow-apis/src/openmetadata/operations/delete.py:46]: it seems like there is a new line after the if condition unless I misreading the formatting here

Comment (pmbrull) [reply openmetadata-airflow-apis/src/openmetadata/operations/delete.py]: ammended

File: openmetadata-airflow-apis/src/openmetadata/api/apis_metadata.py (hunk 2 above)
Comment (pmbrull) [reply openmetadata-airflow-apis/src/openmetadata/api/apis_metadata.py]: updated

**Review by harshach — APPROVED** [review]

**Pull request merged**